Makefile

```diff
@@ -1,6 +1,6 @@
 PROJECT_NAME=datadog-operator
 ARTIFACT=controller
-ARTIFACT_PLUGIN=kubectl-datadog
+ARTIFACT_PLUGIN=datadog
```

Comment:
I'm wondering why do we need this change?

Reply:
that was just because in [all the existing krew manifests](https://krew.sigs.k8s.io/docs/developer-guide/plugin-manifest/) the binaries are named without the `kubectl-` prefix

Reply:
but that's not a problem we can keep the current binary name, anyways the final name is generated from the `metadata.name` field based on https://krew.sigs.k8s.io/docs/developer-guide/plugin-manifest/#specifying-plugin-executable